sdk/core/core-https/documentation/core2.md

```diff
@@ -0,0 +1,69 @@
+# Azure SDK for JS Core 2.0
+
+## What do you mean 2.0?
+
+Today one of the primary core packages we ship is `@azure/core-http`. As of this document being written it sits at version 1.1.5. Due to a number factors discussed in this document, there is a desire to make some breaking changes that would create a new major version of this package (i.e. "2.0.0".)
+
+Since `@azure/core-http` is intended only for indirect use by most consumers, a major version change would not be largely disruptive. Currently, in order to preserve compatibility and not complicate the story for servicing, a new package has been created that is currently named `@azure/core-https`. This name is not final.
+
+## Motivations, Context, and History
+
+In ancient times (~5 years ago), Microsoft provided SDKs for NodeJS developers wanting to interact with Azure services. Ultimately, many of these packages were generated using [AutoRest](https://github.com/Azure/Autorest) using a package called `autorest.nodejs`. The autogenerated code had the need for a common, re-usable core for performing HTTP operations, and thus [`ms-rest`](https://github.com/Azure/ms-rest-js) was created.
+
+Besides `ms-rest` for common code, another package `ms-rest-azure` was created to house Azure-specific components and authetication code that worked with credentials. At this point, support for TypeScript came in the form of hand-authored d.ts files.
```

Comment:
Here, does `common code` apply to `ms-rest` or `ms-rest-azure`? Me confused. How about skipping that part altogether? Extending the creds part with some examples will provide more clarity

```suggestion
Another package `ms-rest-azure` was created to house Azure-specific components and authetication code that worked with a variety of credentials like Service Principal, Device Token, Managed Identity etc. At this point, support for TypeScript came in the form of hand-authored type declaration files.
```